Backend/agent/context.py

```
@@ -1,41 +1,92 @@
+def _format_data_for_embedding(db: sqlite3.Connection) -> tuple[list, list, list]:
+    """
+    Fetches structured data and formats it into individual documents for embedding.
+    """
+    docs, ids, metadatas = [], [], []
 
-        result = "\nAppointments:\n"
-        for a in appointments:
-            result += f"- {a['title']} on {a['appointment_date']} at {a['appointment_time']} ({a['appointment_status']})\n"
+    # Fetch and format appointments
+    appointments = db.execute("SELECT title, appointment_date, appointment_time, appointment_status FROM appointments ORDER BY appointment_date").fetchall()
+    for i, a in enumerate(appointments):
```

Ruff (0.12.2)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)